### ISSUE TITLE
Ignored text actually processed by uncrustify

### ISSUE THREAD

**Issue by dnygren** [uncrustify/uncrustify#2033]
[//]: # "  If the issue is connected to existing Uncrustify options please, if possible, add the"
[//]: # "  following information to ease up the process:"
[//]: # "  • a link to a debug file:"
[//]: # "      generated with: 'uncrustify -p debug.txt -c pathToUsedConfig.cfg -f toBeFormatedFile.cpp' "
[//]: # "      Example: [debug.txt](https://linkToTheFile)"
[//]: # "      Example hosters for debug files: pastbin.com, gist.github.com, ..."
[//]: # "      The used config file is included in the debug file and does not need to be included here."
[//]: # 
[//]: # "  • include a small but complete test file that will be uncrustifyed"
[//]: # "  • include the generated results"
[//]: # "  • include the expected results"
[//]: # 
[//]: # "  ✋ please add a line containing ``` above and below of each of those three code sections"
[//]: # 
[//]: # "  • include the current version of your Uncrustify executable"
[//]: # "      printout via 'uncrustify -v'"
[//]: # "      Example: current version: uncrustify 0.63"
[//]: # "      or if possible additionally with the git sha of the commit"
[//]: # "                    current version: uncrustify 0.63 dc7b412"
[//]: # 
[//]: # "  • if possible include a version that worked"
[//]: # "      Example: working version: uncrustify 0.63"
[//]: # "      or"
[//]: # "                    working version: uncrustify 0.63 2a5e88f"

I use uncrustify to format source code, but often there are sections of header files I want to format other than how I have set up the uncrustify .cfg file. I thought I could just use ```/* *INDENT-OFF* */  /* *INDENT-ON* */``` to turn off uncrustify on those sections of the header file, but it doesn't seem to work all the time. I created a minimal example below where it looks like uncrustify is correctly being told to ignore the section of code I want to remain as is (the debug file even says '''IGNORED''' on those lines), but the uncrustified source and debug output already appears processed by uncrustify.  [Here are the source, .cfg, uncrustified source, and debug files I used at github.](https://github.com/dnygren/uncrustify_debug) What am I doing wrong?
```
$ uncrustify -v
Uncrustify-0.67-485-09a5be76

$ uncrustify -c mmo.cfg -f TEST.hpp -p debug.txt > UNCRUSTIFIED_TEST.hpp
Parsing: TEST.hpp as language CPP

$ sdiff TEST.hpp UNCRUSTIFIED_TEST.hpp
#ifndef TEST_HPP                                                #ifndef TEST_HPP
#define TEST_HPP                                                #define TEST_HPP

/*****************************                                  /*****************************
 * TEST.hpp : Header file test                                   * TEST.hpp : Header file test
 *****************************/                                  *****************************/

/* *INDENT-OFF* */                                              /* *INDENT-OFF* */
enum E_SUNSENSOR {                                            | enum E_SUNSENSOR {    EXAMPLE1,    EXAMPLE2,    SN005};
    EXAMPLE1,                                                 <
    EXAMPLE2,                                                 <
    SN005                                                     <
};                                                            <
/* *INDENT-ON* */                                               /* *INDENT-ON* */

#endif                                                          #endif

$ grep 'IGNORE'  debug.txt
#   9>            IGNORED[               NONE][  1/  1/ 19/  0][0/0/1][     60000][0-0] enum E_SUNSENSOR {
#  10>            IGNORED[               NONE][ 19/  1/ 14/  0][0/0/1][         0][0-0]                       EXAMPLE1,
#  11>            IGNORED[               NONE][ 32/  1/ 14/  0][0/0/1][         0][0-0]                                    EXAMPLE2,
#  12>            IGNORED[               NONE][ 45/  1/ 10/  0][0/0/1][         0][0-0]                                                 SN005
#  13>            IGNORED[               NONE][ 54/  1/  3/  0][0/0/1][         0][0-0]                                                      };
```

**Comment by mihaipopescu** [comment]
Any chance this could have worked in the past and now has regressed with the latest version?

**Comment by CDanU** [comment]
This option is causing this bug:

```cfg
# How aggressively to remove extra newlines not in preproc.
# 0: No change
# 1: Remove most newlines not handled by other config
# 2: Remove all newlines and reformat completely by config
nl_remove_extra_newlines        = 1        # unsigned number
```

**Comment by dnygren** [comment]
> 
> 
> Any chance this could have worked in the past and now has regressed with the latest version?

Mihai -

I started with uncrustify 0.64 that was default on my Debian distro and upgraded to several newer versions in an attempt to resolve the issue, finally pulling the latest (0.67-485-09a5be76) from github.

All of the versions seem to have the same (or at least similar) behavior.

Thanks ... Dan

**Comment by mwoehlke-kitware** [comment]
So... this is fixed by #2034? If so, we can close this, yes? (@CDanU, if a PR fixes an issue, please consider using "Fixes <num>" in the PR description instead of "ref <num>" so the issue will be auto-closed.)

**Comment by dnygren** [comment]
FYI -

I tried Uncrustify-0.67-488-2554a022 that contains the proposed fix without making any modifications to my config file and  ```*INDENT-OFF*``` now properly turns off uncrustify completely until an ```*INDENT-ON*``` is encountered. So it appears you were able to (quickly!) resolve the issue I encountered. Thank you all ... Dan
